Compiler error because of missing include (clang-500.2.79)

Getting errors like this:

```
/Users/mtaylor/nta/nupic/nta/types/Exception.hpp:183:17: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    std::string filename_;
                ^
```

Seems like the version of clang I'm using is stricter than before. It is an easy fix, just add 

```
#include <string>
```

to `Exception.hpp`.

--- a/types/Exception.hpp
+++ b/types/Exception.hpp
@@ -29,6 +29,7 @@
 
 #include <nta/types/types.hpp>
 #include <stdexcept>
+#include <string>
 
 //----------------------------------------------------------------------
 

